### PR TITLE
Create user roles

### PR DESCRIPTION
**Pull request by itsjanevvu**
## Notion Ticket Link

[Create User Roles](https://www.notion.so/uwblueprintexecs/Create-User-Roles-d9bc4346873b4c44964622df3b9563c5?pvs=4)

## Implementation Description
* Added volunteer, staff and admin roles to schema.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```

### CODE REVIEW

**Review by achandrabalan — APPROVED** [review]
LGTM! 
🔥 🧨

**Comment by carolynzhang18** [comment]
Nice work!! Two things:
1. The `role` prop of `AuthenticatedUser` in frontend/src/types/AuthTypes.ts should also be updated.
1.5. some of the unit tests also still have users with role "User", but I'm not sure if we need to update these cuz I don't thiiink we really ever test our code lol - maybe check with Owen on this?
(I did a search (ctrl-shift-F in VSCode) for `"User"` in our codebase to find these places)
2. Remember to run linter on the backend and frontend (command in README)! 🧹 

Otherwise it looks good! 😎

**Review by carolynzhang18 — CHANGES_REQUESTED** [review]
Nice work!! Two things:
1. The `role` prop of `AuthenticatedUser` in frontend/src/types/AuthTypes.ts should also be updated.
1.5. some of the unit tests also still have users with role "User", but I'm not sure if we need to update these cuz I don't thiiink we really ever test our code lol - maybe check with Owen on this?
(I did a search (ctrl-shift-F in VSCode) for `"User"` in our codebase to find these places)
2. Remember to run linter on the backend and frontend (command in README)! 🧹 

Otherwise it looks good! 😎

**Review by carolynzhang18 — APPROVED** [review]
Looks good!! 😻

File: backend/models/user.model.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ const UserSchema: Schema = new Schema({
   role: {
     type: String,
     required: true,
-    enum: ["User", "Admin"],
+    enum: ["Volunteer", "Staff", "Admin"],
   },
 });
 
-export default mongoose.model<User>("User", UserSchema);
+export default mongoose.model<User>("Volunteer", UserSchema);
```
Comment (owen-sellner) [inline backend/models/user.model.ts:33]: This should be left as "User"

**Review by owen-sellner — COMMENTED** [review]
I left one comment but after that it is ready to merge!!!

**Pull request merged**